addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -257,13 +257,15 @@ const Link = Widget.extend({
      * @private
      */
     _correctLink: function (url) {
-        if (url.indexOf('mailto:') === 0 || url.indexOf('tel:') === 0) {
-            url = url.replace(/^tel:([0-9]+)$/, 'tel://$1');
-        } else if (url.indexOf('@') !== -1 && url.indexOf(':') === -1) {
-            url = 'mailto:' + url;
-        } else if (url.indexOf('://') === -1 && url[0] !== '/'
-                    && url[0] !== '#' && url.slice(0, 2) !== '${') {
-            url = 'http://' + url;
+        if (!url.match(/^http[s]?/i)) {
+            if (url.includes('@') && !url.startsWith('mailto:')) {
+                url = 'mailto:' + url;
+            } else if (/^[^a-zA-Z]*[0-9]+[^a-zA-Z]*$/.test(url) && !url.startsWith('tel:')) {
+                url = 'tel:' + url;
+            } else if (!url.includes('://') && url[0] !== '/'
+                        && url[0] !== '#' && url.slice(0, 2) !== '${') {
+                url = 'http://' + url;
```

Comment:
How can I receive an URL which start by http but does not include `://` (and in which case do I want to add http at the start when there is already http at the start)... seems really wrong.